dynamodb/dynamodb-client.w

```diff
@@ -0,0 +1,116 @@
+}
+
+struct Credentials {
+  accessKeyId: str;
+  secretAccessKey: str;
+}
+
+struct CreateDocumentClientOptions {
+  endpoint: str?;
+  region: str?;
+  credentials: Credentials?;
+}
+
+class Util {
+  extern "./dynamodb.mjs" pub static inflight createDocumentClient(options: CreateDocumentClientOptions?): DocumentClient;
+}
+
+pub struct ClientProps {
+  tableName: str;
+  endpoint: str?;
+  region: str?;
+  credentials: Credentials?;
+}
+
+pub inflight class Client impl dynamodb_types.IClient {
+  inflight tableName: str;
+  inflight client: DocumentClient;
+
+  inflight new(props: ClientProps) {
+    this.tableName = props.tableName;
+    this.client = Util.createDocumentClient({
+      region: props.region,
+      credentials: props.credentials,
+      endpoint: props.endpoint,
+    });
+  }
+
+  pub inflight delete(options: dynamodb_types.DeleteOptions): dynamodb_types.DeleteOutput {
+    let input: MutJson = options;
+    input.set("TableName", this.tableName);
+    return unsafeCast(this.client.delete(input));
+  }
+
+  pub inflight get(options: dynamodb_types.GetOptions): dynamodb_types.GetOutput {
+    let input: MutJson = options;
+    input.set("TableName", this.tableName);
+    return unsafeCast(this.client.get(input));
+  }
+
+  pub inflight put(options: dynamodb_types.PutOptions): dynamodb_types.PutOutput {
+    let input: MutJson = options;
+    input.set("TableName", this.tableName);
+    return unsafeCast(this.client.put(input));
+  }
+
+  pub inflight transactWrite(options: dynamodb_types.TransactWriteOptions): dynamodb_types.TransactWriteOutput {
+    let transactItems = MutArray<Json> [];
+    for item in options.TransactItems {
+      if let operation = item.ConditionCheck {
+        let input: MutJson = operation;
+        input.set("TableName", this.tableName);
+        transactItems.push({
+          ConditionCheck: Json.deepCopy(input),
+        });
+      } elif let operation = item.Delete {
+        let input: MutJson = operation;
+        input.set("TableName", this.tableName);
+        transactItems.push({
+          Delete: Json.deepCopy(input),
+        });
+      } elif let operation = item.Put {
+        let input: MutJson = operation;
+        input.set("TableName", this.tableName);
+        transactItems.push({
+          Put: Json.deepCopy(input),
+        });
+      } elif let operation = item.Update {
+        let input: MutJson = operation;
+        input.set("TableName", this.tableName);
+        transactItems.push({
+          Update: Json.deepCopy(input),
+        });
+      } else {
+        throw "Invalid transact item";
+      }
+    }
+    return unsafeCast(this.client.transactWrite({
+      TransactItems: transactItems.copy(),
+    }));
+  }
+
+  pub inflight scan(options: dynamodb_types.ScanOptions?): dynamodb_types.ScanOutput {
+    let input: MutJson = options ?? {};
+    input.set("TableName", this.tableName);
+    return unsafeCast(this.client.scan(input));
```

Comment:
Curious for my own understanding, how tricky is it to implement this method without `unsafeCast`?

Is `let input: MutJson = options ?? {}` a workaround for Winglang not allowing you set struct fields?

Reply:
> It looks like the loop is broken by throwing an error

It is intended. There's a try catch later that ignore certain kind of errors, but throw otherwise.

> how tricky is it to implement this method without unsafeCast?

I guess we could add the proper return types to `DocumentClient` in `dynamodb/dynamodb-client.w` now that they return the original information, but the options still need the `TableName` to be added manually... Any ideas?